src/Stratis.Bitcoin.IntegrationTests/SegWitTests.cs

```diff
@@ -30,44 +28,68 @@ public void TestSegwitActivation()
                 coreNode.ConfigParameters.AddOrReplace("printtoconsole", "0");
                 coreNode.Start();
 
-                CoreNode node1 = builder.CreateStratisPowNode(true, fullNodeBuilder =>
+                CoreNode stratisNode = builder.CreateStratisPowNode(true, fullNodeBuilder =>
                 {
                     fullNodeBuilder
                         .UseConsensus()
                         .UseBlockStore()
+                        .UseWallet()
                         .UseMempool()
-                        .UseBlockNotification()
-                        .UseTransactionNotification()
                         .AddMining()
-                        .UseWallet()                        
-                        .UseApi()
                         .AddRPC();
                 });
 
-                WalletManager wm1 = node1.FullNode.NodeService<IWalletManager>() as WalletManager;
-                wm1.CreateWallet("Test1", "alice1");
-
-                RPCClient rpc1 = node1.CreateRPCClient();
+                RPCClient stratisNodeRpc = stratisNode.CreateRPCClient();
                 RPCClient coreRpc = coreNode.CreateRPCClient();
 
-                coreRpc.AddNode(node1.Endpoint, false);
-                rpc1.AddNode(coreNode.Endpoint, false);
+                coreRpc.AddNode(stratisNode.Endpoint, false);
+                stratisNodeRpc.AddNode(coreNode.Endpoint, false);
 
-                coreRpc.Generate(450);
+                // core (in version 0.15.1) only mines segwit blocks above a certain height on regtest
+                // future versions of core will change that behaviour so this test may need ot be changed in the future
+                // see issue for more details https://github.com/stratisproject/StratisBitcoinFullNode/issues/1028
+                var prevSegwitDeployment = Network.RegTest.Consensus.BIP9Deployments[BIP9Deployments.Segwit];
```

Comment:
unvar

Reply:
I see you've strongly been @Aprogiena-nized